crates/lang/codegen/src/generator/dispatch.rs

```diff
@@ -256,17 +256,22 @@ impl Dispatch<'_> {
                 let constructor_ident = constructor.ident();
                 let selector_id = constructor.composed_selector().into_be_u32().hex_padded_suffixed();
                 let selector_bytes = constructor.composed_selector().hex_lits();
-                let input_bindings = generator::input_bindings(constructor.inputs());
+                let input_bindings = constructor.inputs()
+                    .map(|_| quote! {
+                        ::scale::Decode::decode(&mut _input).map_err(|_| ::ink_lang::reflect::DispatchError::CouldNotReadInput)?
```

Comment:
Also removing of `.map_err` and usage of  `.unwrap()` reduce size for ~500 bytes. So we can create two analogs of `CALLABLE`(for `debug` and `release` mode).

Reply:
`Erc20`: Original wasm size: 31.5K, Optimized: 9.1K(9067) - saved 571
`Erc1155`: Original wasm size: 83.9K, Optimized: 46.2K(46168) - saved 600
`multisig`: Original wasm size: 103.8K, Optimized: 46.8K(46798) - saved 359